calico/etcddriver/driver.py

```diff
@@ -305,6 +313,18 @@ def _resync_and_merge(self):
                 self._resync_requested = False
         _log.info("Stop event set, exiting resync loop.")
 
+    def _rotate_etcd_url(self):
+        """
+        Rotate the in use etcd URL if more than one is configured,
+        """
+        if len(self._etcd_other_urls) > 0:
+            self._etcd_url_lock.acquire()
```

Comment:
You should use `with` whenever dealing with locks unless there's a very good reason not to.
```
with self._etcd_url_lock:
    ...
```
Ensures that the lock is released on all code paths, including exceptions.